pkg/controller/storagecluster/update.go

```diff
@@ -332,6 +337,70 @@ func (c *Controller) getKVDBNodeAvailability(cluster *corev1.StorageCluster) (in
 
 }
 
+func (c *Controller) parallelUpgradeNodesList(cluster *corev1.StorageCluster, pods []*v1.Pod) []*v1.Pod {
+	if len(pods) == 0 {
+		return pods
+	}
+
+	if storagePodsEnabled(cluster) {
+		storageNodeList, err := c.Driver.GetStorageNodes(cluster)
+		if err != nil {
+			logrus.Warnf("Cannot upgrade nodes in parallel, failed to get storage nodes: %v", err)
```

Comment:
Oh like currently we just do not do parallel upgrade but proceed with the normal upgrade in this case. Should we not do the rolling upgrade itself in this case ?

Reply:
yes, let's not proceed in case of error because customer doesn't know here that parallel upgrade is not going to happen. And once nodes are selected for upgrades,we will loose volume quorum and we can't rollback.